shopBuddy/shopBuddy/ShoppingList+Persistent.m

```diff
@@ -0,0 +1,151 @@
+//
+//  ShoppingList+Persistent.m
+//  shopBuddy
+//
+//  Created by Rachna Gupta on 7/21/22.
+//
+
+#import "ShoppingList+Persistent.h"
+#import "Item.h"
+#import "ShoppingList.h"
+#import "Parse/Parse.h"
+#import "Item+Persistent.h"
+#import <objc/runtime.h>
+
+@implementation ShoppingList (Persistent)
+
+- (NSString *)objectID {
+    return objc_getAssociatedObject(self, @selector(objectID));
+}
+
+- (void)setObjectID:(NSString *)new_objectID {
+    objc_setAssociatedObject(self, @selector(objectID), new_objectID, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+- (NSString *)listObject {
+    return objc_getAssociatedObject(self, @selector(listObject));
+}
+
+- (void)setListObject:(NSString *)new_listObject {
+    objc_setAssociatedObject(self, @selector(listObject), new_listObject, OBJC_ASSOCIATION_RETAIN_NONATOMIC);
+}
+
+- (void) _updateSavedListWithNewItem: (PFObject *)item withCompletion:(void(^)(BOOL succeeded, NSError *error))completion {
+    NSMutableArray *previous_items = [NSMutableArray arrayWithArray:self.listObject[@"items"]];
+    [previous_items addObject:item];
+    self.listObject[@"items"] = [NSArray arrayWithArray:previous_items];
+    [self.listObject saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error){
+        if(succeeded) {
+            completion(succeeded, nil);
+        }
+        else {
+            NSLog(@"Error Updating List: %@",error);
+            completion(NO, error);
+        }
+    }];
+}
+
+//Used to create new lists
++ (void) createEmptyList:(NSString *)store_name withCompletion:(void(^)(ShoppingList *new_list,NSError *error))completion {
+    NSDictionary *const dict = @{ @"user" : [PFUser currentUser], @"items" : [NSArray new], @"store_name" : store_name};
+    PFObject *new_object = [PFObject objectWithClassName:@"ShoppingList" dictionary:dict];
+    [new_object saveInBackgroundWithBlock:^(BOOL succeeded, NSError *error){
+        if(succeeded) {
+            ShoppingList *const newList = [ShoppingList _hydrateShoppingListFromPFObject:new_object];
+            completion(newList,nil);
+        }
+        else {
+            NSLog(@"Error: %@",error);
+            completion(nil,error);
+        }
+    }];
+}
+
+//add items to list
++ (void) createFromList:(ShoppingList *)list withItem:(Item *)item withCompletion:(void(^)(BOOL succeeded, NSError *error))completion {
```

Comment:
Shouldn't this method callback return a new copy of `list` with the item added to its item list?